python/cugraph/cugraph/gnn/data_loading/bulk_sampler.py

```diff
@@ -193,7 +193,11 @@ def flush(self) -> None:
         if isinstance(self.__batches, dask_cudf.DataFrame):
             self.__batches = self.__batches.persist()
 
-        min_batch_id = int(self.__batches[self.batch_col_name].min().compute())
+        min_batch_id = self.__batches[self.batch_col_name].min()
+        if isinstance(self.__batches, dask_cudf.DataFrame):
+            min_batch_id = min_batch_id.compute()
+        min_batch_id = int(min_batch_id)
```

Comment:
Why did we add this compute back ?

Reply:
`client.submit` doesn't accept it uncomputed.  What we had before worked with `client.compute`.  I'm trying to find alternative that works for `submit`.

The error we get is this:
```
AttributeError("'Scalar' object has no attribute '_parent_meta'")
```

And wrapping with delayed doesn't work either, it gives a similar error.

Reply:
i.e.
`delayed(lambda df : df.min())(self.__batches)` doesn't work

Reply:
I think I found a way around that issue, but we have another problem where the execution hangs because we are setting allow_other_workers=False so it can't get the min

Reply:
I guess we can let it be for now and try to get other optimizations as a followup

Reply:
I've since figured out that what we had with `client.compute` and not computing the min beforehand was wrong.  It should cause a hang, which we were seeing.

Reply:
#3690